examples/sst2/train.py

```diff
@@ -42,23 +50,35 @@ def sigmoid_cross_entropy_with_logits(*, labels: Array, logits: Array) -> Array:
   return relu_logits - logits * labels + jnp.log1p(jnp.exp(neg_abs_logits))
 
 
-def get_initial_params_and_state(key, model):
+def get_initial_params_and_state(rng, model):
   """Returns randomly initialized parameters and a fresh model state."""
   token_ids = jnp.ones((2, 3), jnp.int32)
   lengths = jnp.ones((2,), dtype=jnp.int32)
-  variables = model.init(key, token_ids, lengths)
-  state, params = variables.pop('params')
-  return params, state
+  variables = model.init(rng, token_ids, lengths, deterministic=True)
+  model_state, params = variables.pop('params')
```

Comment:
What about keeping individual variable collections in separate variables?
I find that more readable, especially when there are `intermediates` lurking around.

Like we do here:
https://github.com/google/flax/blob/0082b90c36514b4947834cafc99172ec91ed16b5/examples/imagenet/train.py#L72

Reply:
Done

Reply:
Did you push? I'm still seeing `model_state`...

Reply:
Ah it was only about `intermediates` in the first place... I tend more and more to always spelling them out because having that `variables` dict is just very opaque.